### PR TITLE
Attempt to split attribute by type

### PR DESCRIPTION
**Pull request by martinthomson**
HTML vs. IDL

***
<a href="https://pr-preview.s3.amazonaws.com/martinthomson/design-principles/pull/485.html" title="Last updated on Jun 3, 2024, 3:57 PM UTC (4775ff0)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3ctag/design-principles/485/222226d...martinthomson:4775ff0.html" title="Last updated on Jun 3, 2024, 3:57 PM UTC (4775ff0)">Diff</a>

### CODE REVIEW

File: index.bs [hunk 1]
```diff
@@ -1449,37 +1449,36 @@ See also:
 
 * [[#attributes-like-data]]
 
-<h3 id="attributes-vs-methods">Use attributes or methods appropriately</h3>
+<h3 id="attributes-vs-methods">Properties, getters, and setters should behave like properties, not methods</h3>
```
Comment (LeaVerou) [inline index.bs:1452]:
```suggestion
<h3 id="attributes-vs-methods">Accessors should behave like properties, not methods</h3>
```

The correct name for "getters and setters" is "accessors"

File: index.bs [hunk 2]
```diff
@@ -1489,7 +1488,7 @@ This means:
 
 - If live, then return the same object each time,
     until a state change requires a different object to be returned.
-    This can be returned from either an attribute or a method.
+    This can be returned from either an property, getter, or method.
```
Comment (martinthomson) [inline index.bs:1491]:
```suggestion
    This can be returned from either a property, getter, or method.
```

**Comment by annevk** [comment]
The HTML Standard uses "content attributes" and "IDL attributes". Not a big fan of that terminology, but at least you don't rule out SVG and MatML immediately.

**Comment by LeaVerou** [comment]
> The HTML Standard uses "content attributes" and "IDL attributes". Not a big fan of that terminology, but at least you don't rule out SVG and MatML immediately.

Ooh good point, we definitely don't want to exclude those.
Some other ideas:
- markup attributes
- element attributes

**Review by rhiaro — APPROVED** [review]

**Comment by martinthomson** [comment]
@LeaVerou is there a problem with "HTML attribute" in the context that it appears?  @rhiaro suggests that this is an improvement, even if it isn't perfect.

File: index.bs [hunk 3]
```diff
@@ -1449,37 +1449,36 @@ See also:
...
+- A getter must not have any (observable) side effects.
     If you have expected side effects, use a method.
 
-- Attribute getters should not throw exceptions.
+- Getters should not throw exceptions.
     Getters should [behave like regular data properties](#attributes-like-data), and regular data properties do not throw exceptions when read. Furthermore, invalid state should generally be avoided by rejecting *writes*, not when data is *read*.
     Updating existing getters to throw exceptions should be avoided as existing API users may enumerate or wrap the API and not expect the new exception, breaking backwards compatibility.
 
-- Attribute getters should not perform any blocking operations.
+-  Getters should not perform any blocking operations.
```
Comment (martinthomson) [inline index.bs:1463]:
```suggestion
- Getters should not perform any blocking operations.
```

**Review by plinss — APPROVED** [review]

**Pull request merged**